server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -523,7 +533,24 @@ public Indexer(String indexName,
                 indexColumns.add(new IndexColumn(ref.column(), fieldType, indexInputs));
             }
         }
-        this.returnValues = returnValues;
+        if (returnValues == null) {
+            this.returnValueInputs = null;
+        } else {
+            this.returnValueInputs = new ArrayList<>(returnValues.length);
+            for (int i = 0; i < returnValues.length; i++) {
+                Symbol updatedReturnValue = RefReplacer.replaceRefs(returnValues[i], ref -> {
+                    Synthetic synthetic = synthetics.get(ref.column());
+                    if (synthetic != null) {
+                        return Literal.ofUnchecked(ref.valueType(), synthetic.value());
```

Comment:
Doesn't this freeze the value at construction time? 
We'd need to get the `returnValueInputs` to call into the synthethic. That means either we've to override the `Literal#value` call here, or we'd have to change the `InputFactory` stuff to return different inputs for the functions.

Reply:
Yep, I have something different locally, created anther factory to either get synthetic or getImplementation

Reply:
Replied in https://github.com/crate/crate/pull/14762#discussion_r1342532437